erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -2023,15 +2034,25 @@ func (hph *HexPatriciaHashed) Process(ctx context.Context, updates *Updates, log
 		}
 		// Keep folding until the currentKey is the prefix of the key we modify
 		for hph.needFolding(hashedKey) {
+			startFold := time.Now()
```

Comment:
timer start should be controlled by fact of flag provision to avoid unnecessary allocations when not needed